watsononlinestore/watson_online_store.py

```diff
@@ -292,6 +294,23 @@ def get_discovery_response(self, input_text):
             query_options={'query': input_text, 'count': DISCOVERY_QUERY_COUNT}
         )
 
+        # Watson discovery assigns a confidence level to each result.
+        # Based on data mix, we can assign a minimum tolerance value in an
+        # attempt to filter out the "weakest" results.
+        num_matches = 0
+        filtered_results = []
+        if self.discovery_score_filter:
+            results = (results for results in discovery_response['results']
+                      if 'results' in discovery_response)
```

Comment:
OK there are some things inside the loop that could be done and we return a list so this might be a good thing to do:
        filtered_results = []
        if self.discovery_score_filter:
            filtered_results = [r for r in discovery_response.get('results', []) \
                                if ('score' in r and r['score'] >= self.discovery_score_filter)]
        num_matches = len(filtered_results)